habitat-baselines/habitat_baselines/rl/ddppo/policy/resnet_policy.py

```diff
@@ -326,6 +348,53 @@ def __init__(
             )
             rnn_input_size += 32
 
+        if ObjectEmbeddingSensor.cls_uuid in observation_space.spaces:
```

Comment:
Pass these via the fuse_keys in the config so you don't have to hand define these sensors as input.

Reply:
While ObjectEmbeddingSensor, which is a frozen CLIP embedding, can be passed as fuse_keys, the categorical inputs cannot be. The categorical inputs learn an embedding dictionary which needs to be stored in PointNavResNetPolicy, requiring this additional handling. For this, I just followed the handling of the ObjectGoal sensor that ObjectNav uses. I believe eventually we will be using frozen-embedding-based sensors for objects/receptacles so we will remove the categorical sensors altogether in future. I am keeping them now for experimentation.

For ObjectEmbeddingSensor, I will switch to fuse_keys as you suggested.

Reply:
I see, that makes sense.